app/src/main/java/net/gsantner/markor/ui/fsearch/SearchEngine.java

```diff
@@ -429,5 +439,24 @@ private boolean isIgnored(File directory) {
             }
             return false;
         }
+
+        private InputStream getInputStream(File file) throws FileNotFoundException {
+            if (isEncryptedFile(file)) {
+                final byte[] encryptedContext = FileUtils.readCloseStreamWithSize(new FileInputStream(file), (int) file.length());
+                if (encryptedContext.length > JavaPasswordbasedCryption.Version.NAME_LENGTH && _config.password.length > 0) {
+                    final byte[] decrypt = new JavaPasswordbasedCryption(getVersion(encryptedContext), null).decryptBytes(encryptedContext, _config.password.clone());
+                    return new ByteArrayInputStream(decrypt);
+                } else {
+                    return new ByteArrayInputStream(encryptedContext);
+                }
```

Comment:
don't we have a function for this already somewhere. looks a lot like duplicated code here again

Reply:
Not exactly, but it was a form of pre optimization. Fixed